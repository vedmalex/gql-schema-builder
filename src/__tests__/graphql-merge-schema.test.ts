import 'jest'
import gql from 'graphql-tag'
import mergeTypes from '../graphql-merge-schema'

describe('Merger', () => {
  it('merge classes', () => {
    const type1 = gql`
      type Query {
        login(user: String, password: String): Boolean @acl(level: ANONYMOUS)
      }
    `
    const directive = gql`
      directive @acl(
        level: [Acl] = [AUTHENTICATED]
      ) on OBJECT | FIELD_DEFINITION
    `
    const type2 = gql`
      type Query {
        logout(user: String): Boolean @acl(level: AUTHENTICATED)
      }
    `
    const schema = mergeTypes([type1, type2, directive])
    expect(schema).toMatchSnapshot('schema')
  })

  it('Merges UNIONS', () => {
    const Acl1 = gql`
      enum ACL {
        ADMIN
        AUTHENTICATED
      }
    `
    const Acl2 = gql`
      enum ACL {
        STUDENT
        EXAMINTOR
        VIEWER
      }
    `
    const schema = mergeTypes([Acl1, Acl2])
    expect(schema).toMatchSnapshot('snap')
  })
})
