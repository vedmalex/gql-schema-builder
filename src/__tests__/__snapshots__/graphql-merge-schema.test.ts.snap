// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Merger Merges UNIONS: snap 1`] = `
"enum ACL {
  ADMIN
  AUTHENTICATED
  STUDENT
  EXAMINTOR
  VIEWER
}
"
`;

exports[`Merger merge classes: schema 1`] = `
"type Query {
  login(user: String, password: String): Boolean @acl(level: ANONYMOUS)
  logout(user: String): Boolean @acl(level: AUTHENTICATED)
}

directive @acl(level: [Acl] = [AUTHENTICATED]) on OBJECT | FIELD_DEFINITION
"
`;
